enclave/core/sgx/globals.c

```diff
@@ -213,8 +222,8 @@ const void* __oe_get_heap_base()
 size_t __oe_get_heap_size()
 {
 #ifdef OE_WITH_EXPERIMENTAL_EEID
-    if (oe_eeid)
-        return oe_eeid->size_settings.num_heap_pages * OE_PAGE_SIZE;
+    if (oe_have_eeid())
+        return oe_get_eeid()->size_settings.num_heap_pages * OE_PAGE_SIZE;
```

Comment:
"Static size" EEID enclave's base image already specifies heap/stack size and number of threads in `oe_enclave_properties_sgx`. If an malicious loader sets `size_settings` in the eeid page differently, the enclave code might be fooled to read/write into unepxected memeory location. Whether the attack can be used to manipulate the eeid data inlcuded in the Attestation Evidence to still pass the verification requires more analysis. It's safer to use the heap size info in the base enclave image in the "static size" EEID enclave case. The 0-heap/0-stack/0-TCS flag of the "dynamic size" EEID enclave can be used to switch the code between `size_settings` in `oe_enclave_properties_sgx` and  `size_settings` in the eeid page.

Reply:
Outdated.